### ISSUE TITLE
please help: ing-diba not working any longer (locks account)

### ISSUE THREAD

**Issue by prankousky** [bnw/firefly-iii-fints-importer#42]
Hi, 

I thought that I should mention this... my `ing-diba` account has been locked due to >3 incorrect logins. I didn't know what to make of it, until I realized that `firefly-iii-fints-importer` logs into my bank as well.

It used to work fine, but suddenly started displaying this error message

```
Fatal error: Uncaught Fhp\Protocol\ServerException: FinTS errors: 9800 (global): Der Dialog wurde abgebrochen. 9931 (global): Zugang ist gesperrt. Tel. Entsperrung: 069 / 34 22 24 (nur mit Telebanking PIN). in /vendor/nemiah/php-fints/lib/Fhp/Protocol/ServerException.php:196 Stack trace: #0 /vendor/nemiah/php-fints/lib/Fhp/FinTs.php(925): Fhp\Protocol\ServerException::detectAndThrowErrors(Object(Fhp\Protocol\Message), Object(Fhp\Protocol\Message)) #1 /vendor/nemiah/php-fints/lib/Fhp/FinTs.php(353): Fhp\FinTs->sendMessage(Object(Fhp\Protocol\Message)) #2 /vendor/nemiah/php-fints/lib/Fhp/FinTs.php(312): Fhp\FinTs->sendRequestForAction(Object(Fhp\Protocol\DialogInitialization), Object(Fhp\Protocol\Message)) #3 /vendor/nemiah/php-fints/lib/Fhp/FinTs.php(789): Fhp\FinTs->execute(Object(Fhp\Protocol\DialogInitialization)) #4 /vendor/nemiah/php-fints/lib/Fhp/FinTs.php(632): Fhp\FinTs->executeWeakDialogInitialization(NULL) #5 /vendor/nemiah/php-fints/lib/Fhp/FinTs.php(258): Fhp\FinTs->ensureSynchronized() #6 /app/index.php(130): in /vendor/nemiah/php-fints/lib/Fhp/Protocol/ServerException.php on line 196
```

The first time this happened, I didn't realize each importer error message meant one incorrect login, so I kept trying. In fact, I kept trying to much that my banking account was blocked permanently (I could unlock it via phone banking, but **could not** gain access to the unlocked account because none of the user credentials would be valid any longer). 

I received new credentials via snail mail today. Finally, I can access my bank account again. However, as soon as I started trying importer again, the same error happened.

Configuration

```
{
  "bank_username": "last 10 digits of my account number",
  "bank_password": "bank login password",
  "bank_code": "50010517",
  "bank_url": "https://fints.ing.de/fints/",
  "bank_2fa": "NoPsd2TanMode",
  "firefly_url": "firefly iii url",
  "firefly_access_token": firefly iii access token"
}
```

It **seems** like they changed the bank url from https://fints.ing-diba.de/fints/ to https://fints.ing.de/fints/. I changed this in my configuration, but no success. Also, my bank password is definitely correct. I had to change it multiple times today (as my account got blocked right again after trying to connect importer), saved it to my password manager each time, and copy/pasted it to the config. So there is no typo there. Also, due to changing it multiple times, I had to always enter it in my browser and smartphone as well - and there, the password worked.

Question: do I need to provide the mobile pin instead of the actual password to importer? I am afraid to try this now, as last time I tried things lead to my account getting locked to the point where I could not manually unlock it and had to wait for new credentials.

I love importer, and firefly iii is useless to me without it. Is there anything I missed? Thank you in advance for your help :)

**Comment by bnw** [comment]
Hi,
unfortunately, as I don't have any account with ing-diba, I can only give some general advice.

>It used to work fine, but suddenly started displaying this error message  

Some googling revealed https://www.willuhn.de/wiki/doku.php?id=psd2#ing
This mentions that a login in the web app is required every 90 days. Could that have been the problem here?

Additionally, the wiki suggests, that for ING only FINTS 2.2 works. However, this is contradicting to the information [here](https://www.ing.de/hilfe/psd2/kundenservice/).
The FINTS library this importer uses only supports FINTS 3.0, so testing this is not really an option anyway.

https://www.ing.de/hilfe/psd2/kundenservice/ also explains what password should be used.

My last idea would be to get in contact with the bank direkly. Possibly, FINTS is still disabled on your account for some reason?

**Comment by prankousky** [comment]
Thank you and huge sorry for this: I was permanently working with the wrong `.json` file. I updated it when I changed my PIN, but importer was looking at a **different** file (well, same file name, different path), so it *kept* attempting to log in with the old password. Hence the blocking of my account multiple times.

I realized this when I created a test.json, restarted the importer container, and this file wasn't there. So this issue was all for nothing. There is nothing wrong with importer, or the connection between it and ING, I just didn't edit the correct file and caused this problem all by myself. Verplantheit lässt grüßen.
